src/txn/txn_rollback_to_stable.c

```diff
@@ -998,6 +1016,74 @@ __rollback_to_stable_btree_hs_truncate(WT_SESSION_IMPL *session, uint32_t btree_
     return (ret);
 }
 
+/*
+ * __rollback_to_stable_hs_final_pass --
+ *     Perform rollback to stable on the history store to remove any updates more than stable
+ *     timestamp.
+ */
+static int
+__rollback_to_stable_hs_final_pass(WT_SESSION_IMPL *session, wt_timestamp_t rollback_timestamp)
+{
+    WT_CONFIG ckptconf;
+    WT_CONFIG_ITEM cval, durableval, key;
+    WT_DECL_RET;
+    wt_timestamp_t max_durable_ts, newest_start_durable_ts, newest_stop_durable_ts;
+    char *config;
+    char ts_string[2][WT_TS_INT_STRING_SIZE];
+    bool durable_ts_found;
+
+    config = NULL;
+
+    WT_RET(__wt_metadata_search(session, WT_HS_URI, &config));
+    /* Find out the max durable timestamp of the object from checkpoint. */
+    newest_start_durable_ts = newest_stop_durable_ts = WT_TS_NONE;
+    durable_ts_found = false;
+    WT_ERR(__wt_config_getones(session, config, "checkpoint", &cval));
+    __wt_config_subinit(session, &ckptconf, &cval);
+    for (; __wt_config_next(&ckptconf, &key, &cval) == 0;) {
+        ret = __wt_config_subgets(session, &cval, "newest_start_durable_ts", &durableval);
+        if (ret == 0) {
+            newest_start_durable_ts =
+              WT_MAX(newest_start_durable_ts, (wt_timestamp_t)durableval.val);
+            durable_ts_found = true;
+        }
+        WT_ERR_NOTFOUND_OK(ret, false);
+        ret = __wt_config_subgets(session, &cval, "newest_stop_durable_ts", &durableval);
+        if (ret == 0) {
+            newest_stop_durable_ts = WT_MAX(newest_stop_durable_ts, (wt_timestamp_t)durableval.val);
+            durable_ts_found = true;
+        }
+        WT_ERR_NOTFOUND_OK(ret, false);
+    }
+    max_durable_ts = WT_MAX(newest_start_durable_ts, newest_stop_durable_ts);
+    WT_ERR(__wt_session_get_dhandle(session, WT_HS_URI, NULL, NULL, 0));
+
+    /*
+     * The rollback operation should be performed on this file based on the following:
+     * 1. The tree is modified.
+     * 2. The checkpoint durable start/stop timestamp is greater than the rollback timestamp.
+     * 3. There is no durable timestamp in any checkpoint.
+     */
+    if (S2BT(session)->modified || max_durable_ts > rollback_timestamp || !durable_ts_found) {
```

Comment:
for which case do we require, the condition `!durable_ts_found` only?  I think that condition is redundant.

Reply:
Yes, those all conditions are not required. Just the timestamp check is enough. Fixed.